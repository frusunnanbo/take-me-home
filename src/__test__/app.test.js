const request = require('supertest');
const nock = require('nock');
const fs = require('fs');
const app = require('../app');

describe('The Take-me-home app', () => {

  jest.setTimeout(60000);

  beforeAll(() => {
    nock('https://api.resrobot.se')
      .get(/\/v2\/trip/)
      .reply(200, JSON.parse(fs.readFileSync('resrobot-response.json')));
  });

  it('Returns some trips', (done) => {

    request(app)
      .get('/?hereLat=56.4883233&hereLong=16.3842807&homeLat=59.2733699&homeLong=18.0183112&startTime=2019-08-04T19:02')
      .expect(200)
      .end((err, res) => {
        expect(res.body).toMatchSnapshot();
        done();
      });

  });

});
