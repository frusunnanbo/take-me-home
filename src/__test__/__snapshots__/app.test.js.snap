// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Take-me-home app Returns some trips 1`] = `
Array [
  Object {
    "start": Object {
      "date": "2019-08-05",
      "time": "22:02:00",
    },
    "stops": Array [
      "Hedvigsborg (Mörbylånga kn)",
      "Resmo vsk (Mörbylånga kn)",
      "Kleva (Mörbylånga kn)",
      "Lilla Frö (Mörbylånga kn)",
      "Stora Frö (Mörbylånga kn)",
      "Haga park (Mörbylånga kn)",
      "Vickleby vsk (Mörbylånga kn)",
      "Karlevi vsk (Mörbylånga kn)",
      "Eriksöre camping (Mörbylånga kn)",
      "Runsbäck (Mörbylånga kn)",
      "Färjestaden Byggmästargatan (Mörbylånga kn)",
      "Södra Hamnplan (Mörbylånga kn)",
      "Färjestaden Ölands Köpstad (Mörbylånga kn)",
      "Snäckstrand södra (Mörbylånga kn)",
      "Snäckstrand norra (Mörbylånga kn)",
      "Färjestaden Möllstorp (Mörbylånga kn)",
      "Tallhagen (Kalmar kn)",
      "Kalmar Chokladfabriken",
      "Kalmar Malmen",
      "Tullslätten (Kalmar kn)",
      "Kalmar Centralstation",
      "Mönsterås norra",
      "Oskarshamn station",
      "Västervik station",
      "Norrköping busstation",
      "Kolmården station (Norrköping kn)",
      "Nyköping Centralstation",
      "Vagnhärad station (Trosa kn)",
      "Södertälje Syd station",
      "Flemingsberg station (Huddinge kn)",
      "Huddinge station",
      "Stuvsta station (Huddinge kn)",
      "Älvsjö station (Stockholm kn)",
    ],
  },
  Object {
    "start": Object {
      "date": "2019-08-05",
      "time": "22:02:00",
    },
    "stops": Array [
      "Hedvigsborg (Mörbylånga kn)",
      "Resmo vsk (Mörbylånga kn)",
      "Kleva (Mörbylånga kn)",
      "Lilla Frö (Mörbylånga kn)",
      "Stora Frö (Mörbylånga kn)",
      "Haga park (Mörbylånga kn)",
      "Vickleby vsk (Mörbylånga kn)",
      "Karlevi vsk (Mörbylånga kn)",
      "Eriksöre camping (Mörbylånga kn)",
      "Runsbäck (Mörbylånga kn)",
      "Färjestaden Byggmästargatan (Mörbylånga kn)",
      "Södra Hamnplan (Mörbylånga kn)",
      "Färjestaden Ölands Köpstad (Mörbylånga kn)",
      "Snäckstrand södra (Mörbylånga kn)",
      "Snäckstrand norra (Mörbylånga kn)",
      "Färjestaden Möllstorp (Mörbylånga kn)",
      "Tallhagen (Kalmar kn)",
      "Kalmar Chokladfabriken",
      "Kalmar Malmen",
      "Tullslätten (Kalmar kn)",
      "Kalmar Centralstation",
      "Mönsterås norra",
      "Oskarshamn station",
      "Västervik station",
      "Norrköping busstation",
      "Kolmården station (Norrköping kn)",
      "Nyköping Centralstation",
      "Vagnhärad station (Trosa kn)",
      "Södertälje Syd station",
      "Flemingsberg station (Huddinge kn)",
      "Huddinge station",
      "Stuvsta station (Huddinge kn)",
      "Älvsjö station (Stockholm kn)",
      "Turingevägen (Stockholm kn)",
    ],
  },
  Object {
    "start": Object {
      "date": "2019-08-05",
      "time": "22:02:00",
    },
    "stops": Array [
      "Hedvigsborg (Mörbylånga kn)",
      "Resmo vsk (Mörbylånga kn)",
      "Kleva (Mörbylånga kn)",
      "Lilla Frö (Mörbylånga kn)",
      "Stora Frö (Mörbylånga kn)",
      "Haga park (Mörbylånga kn)",
      "Vickleby vsk (Mörbylånga kn)",
      "Karlevi vsk (Mörbylånga kn)",
      "Eriksöre camping (Mörbylånga kn)",
      "Runsbäck (Mörbylånga kn)",
      "Färjestaden Byggmästargatan (Mörbylånga kn)",
      "Södra Hamnplan (Mörbylånga kn)",
      "Färjestaden Ölands Köpstad (Mörbylånga kn)",
      "Snäckstrand södra (Mörbylånga kn)",
      "Snäckstrand norra (Mörbylånga kn)",
      "Färjestaden Möllstorp (Mörbylånga kn)",
      "Tallhagen (Kalmar kn)",
      "Kalmar Chokladfabriken",
      "Kalmar Malmen",
      "Tullslätten (Kalmar kn)",
      "Kalmar Centralstation",
      "Mönsterås norra",
      "Oskarshamn station",
      "Västervik station",
      "Norrköping busstation",
      "Stockholm Cityterminalen",
      "Stockholm Södra station",
      "Årstaberg station (Stockholm kn)",
      "Älvsjö station (Stockholm kn)",
    ],
  },
  Object {
    "start": Object {
      "date": "2019-08-05",
      "time": "22:02:00",
    },
    "stops": Array [
      "Hedvigsborg (Mörbylånga kn)",
      "Resmo vsk (Mörbylånga kn)",
      "Kleva (Mörbylånga kn)",
      "Lilla Frö (Mörbylånga kn)",
      "Stora Frö (Mörbylånga kn)",
      "Haga park (Mörbylånga kn)",
      "Vickleby vsk (Mörbylånga kn)",
      "Karlevi vsk (Mörbylånga kn)",
      "Eriksöre camping (Mörbylånga kn)",
      "Runsbäck (Mörbylånga kn)",
      "Färjestaden Byggmästargatan (Mörbylånga kn)",
      "Södra Hamnplan (Mörbylånga kn)",
      "Färjestaden Ölands Köpstad (Mörbylånga kn)",
      "Snäckstrand södra (Mörbylånga kn)",
      "Snäckstrand norra (Mörbylånga kn)",
      "Färjestaden Möllstorp (Mörbylånga kn)",
      "Tallhagen (Kalmar kn)",
      "Kalmar Chokladfabriken",
      "Kalmar Malmen",
      "Tullslätten (Kalmar kn)",
      "Kalmar Centralstation",
      "Mönsterås norra",
      "Oskarshamn station",
      "Västervik station",
      "Norrköping busstation",
      "Stockholm Cityterminalen",
      "Stockholm Södra station",
      "Årstaberg station (Stockholm kn)",
      "Älvsjö station (Stockholm kn)",
      "Turingevägen (Stockholm kn)",
    ],
  },
  Object {
    "start": Object {
      "date": "2019-08-06",
      "time": "05:02:00",
    },
    "stops": Array [
      "Hedvigsborg (Mörbylånga kn)",
      "Resmo vsk (Mörbylånga kn)",
      "Kleva (Mörbylånga kn)",
      "Lilla Frö (Mörbylånga kn)",
      "Stora Frö (Mörbylånga kn)",
      "Haga park (Mörbylånga kn)",
      "Vickleby vsk (Mörbylånga kn)",
      "Karlevi vsk (Mörbylånga kn)",
      "Eriksöre camping (Mörbylånga kn)",
      "Runsbäck (Mörbylånga kn)",
      "Färjestaden Byggmästargatan (Mörbylånga kn)",
      "Södra Hamnplan (Mörbylånga kn)",
      "Färjestaden Ölands Köpstad (Mörbylånga kn)",
      "Snäckstrand södra (Mörbylånga kn)",
      "Snäckstrand norra (Mörbylånga kn)",
      "Färjestaden Möllstorp (Mörbylånga kn)",
      "Tallhagen (Kalmar kn)",
      "Kalmar Chokladfabriken",
      "Kalmar Malmen",
      "Tullslätten (Kalmar kn)",
      "Kalmar Centralstation",
      "Mönsterås norra",
      "Oskarshamn station",
      "Fårbo E22 (Oskarshamn kn)",
      "Misterhult vägskäl E22 (Oskarshamn kn)",
      "Blankaholm E22 (Västervik kn)",
      "Verkebäcksbron (Västervik kn)",
      "Västervik station",
      "Gamleby Flisvägen (Västervik kn)",
      "Valdemarsvik E22",
      "Söderköping busstation",
      "Norrköping busstation",
      "Nyköping busstation",
      "Stockholm Cityterminalen",
      "Stockholm Södra station",
      "Årstaberg station (Stockholm kn)",
      "Älvsjö station (Stockholm kn)",
    ],
  },
  Object {
    "start": Object {
      "date": "2019-08-06",
      "time": "05:02:00",
    },
    "stops": Array [
      "Hedvigsborg (Mörbylånga kn)",
      "Resmo vsk (Mörbylånga kn)",
      "Kleva (Mörbylånga kn)",
      "Lilla Frö (Mörbylånga kn)",
      "Stora Frö (Mörbylånga kn)",
      "Haga park (Mörbylånga kn)",
      "Vickleby vsk (Mörbylånga kn)",
      "Karlevi vsk (Mörbylånga kn)",
      "Eriksöre camping (Mörbylånga kn)",
      "Runsbäck (Mörbylånga kn)",
      "Färjestaden Byggmästargatan (Mörbylånga kn)",
      "Södra Hamnplan (Mörbylånga kn)",
      "Färjestaden Ölands Köpstad (Mörbylånga kn)",
      "Snäckstrand södra (Mörbylånga kn)",
      "Snäckstrand norra (Mörbylånga kn)",
      "Färjestaden Möllstorp (Mörbylånga kn)",
      "Tallhagen (Kalmar kn)",
      "Kalmar Chokladfabriken",
      "Kalmar Malmen",
      "Tullslätten (Kalmar kn)",
      "Kalmar Centralstation",
      "Mönsterås norra",
      "Oskarshamn station",
      "Fårbo E22 (Oskarshamn kn)",
      "Misterhult vägskäl E22 (Oskarshamn kn)",
      "Blankaholm E22 (Västervik kn)",
      "Verkebäcksbron (Västervik kn)",
      "Västervik station",
      "Gamleby Flisvägen (Västervik kn)",
      "Valdemarsvik E22",
      "Söderköping busstation",
      "Norrköping busstation",
      "Nyköping busstation",
      "Stockholm Cityterminalen",
      "Stockholm Södra station",
      "Årstaberg station (Stockholm kn)",
      "Älvsjö station (Stockholm kn)",
      "Turingevägen (Stockholm kn)",
    ],
  },
  Object {
    "start": Object {
      "date": "2019-08-06",
      "time": "06:02:00",
    },
    "stops": Array [
      "Hedvigsborg (Mörbylånga kn)",
      "Resmo vsk (Mörbylånga kn)",
      "Kleva (Mörbylånga kn)",
      "Lilla Frö (Mörbylånga kn)",
      "Stora Frö (Mörbylånga kn)",
      "Haga park (Mörbylånga kn)",
      "Vickleby vsk (Mörbylånga kn)",
      "Karlevi vsk (Mörbylånga kn)",
      "Eriksöre camping (Mörbylånga kn)",
      "Runsbäck (Mörbylånga kn)",
      "Färjestaden Byggmästargatan (Mörbylånga kn)",
      "Södra Hamnplan (Mörbylånga kn)",
      "Färjestaden Ölands Köpstad (Mörbylånga kn)",
      "Snäckstrand södra (Mörbylånga kn)",
      "Snäckstrand norra (Mörbylånga kn)",
      "Färjestaden Möllstorp (Mörbylånga kn)",
      "Tallhagen (Kalmar kn)",
      "Kalmar Chokladfabriken",
      "Kalmar Malmen",
      "Tullslätten (Kalmar kn)",
      "Kalmar Centralstation",
      "Nybro station",
      "Emmaboda station",
      "Lessebo station",
      "Hovmantorp station (Lessebo kn)",
      "Växjö station",
      "Alvesta station",
      "Nässjö Centralstation",
      "Mjölby station",
      "Linköping Centralstation",
      "Norrköping Centralstation",
      "Stockholm Södra station",
      "Årstaberg station (Stockholm kn)",
      "Älvsjö station (Stockholm kn)",
    ],
  },
  Object {
    "start": Object {
      "date": "2019-08-06",
      "time": "06:02:00",
    },
    "stops": Array [
      "Hedvigsborg (Mörbylånga kn)",
      "Resmo vsk (Mörbylånga kn)",
      "Kleva (Mörbylånga kn)",
      "Lilla Frö (Mörbylånga kn)",
      "Stora Frö (Mörbylånga kn)",
      "Haga park (Mörbylånga kn)",
      "Vickleby vsk (Mörbylånga kn)",
      "Karlevi vsk (Mörbylånga kn)",
      "Eriksöre camping (Mörbylånga kn)",
      "Runsbäck (Mörbylånga kn)",
      "Färjestaden Byggmästargatan (Mörbylånga kn)",
      "Södra Hamnplan (Mörbylånga kn)",
      "Färjestaden Ölands Köpstad (Mörbylånga kn)",
      "Snäckstrand södra (Mörbylånga kn)",
      "Snäckstrand norra (Mörbylånga kn)",
      "Färjestaden Möllstorp (Mörbylånga kn)",
      "Tallhagen (Kalmar kn)",
      "Kalmar Chokladfabriken",
      "Kalmar Malmen",
      "Tullslätten (Kalmar kn)",
      "Kalmar Centralstation",
      "Nybro station",
      "Emmaboda station",
      "Lessebo station",
      "Hovmantorp station (Lessebo kn)",
      "Växjö station",
      "Alvesta station",
      "Nässjö Centralstation",
      "Mjölby station",
      "Linköping Centralstation",
      "Norrköping Centralstation",
      "Stockholm Södra station",
      "Årstaberg station (Stockholm kn)",
      "Älvsjö station (Stockholm kn)",
      "Turingevägen (Stockholm kn)",
    ],
  },
]
`;
